ext/setup/theme.php

```diff
@@ -99,4 +99,36 @@ protected function sb_to_html(SetupBlock $block): string
 		";
         return $html;
     }
+
+    public function format_option(string $name, $html, ?string $label, bool $table_row, bool $label_row = false): string
+    {
+        $output = "";
+        if ($table_row) {
+            $output .= "<tr><th colspan='".($label_row ? 2 : 1)."' style='text-align: ".($label_row ? 'center' : 'right')."'>";
+        }
+        if (!is_null($label)) {
+            $output .= "<label for='{$name}'>{$label}</label>";
+        }
+
+        if ($table_row) {
+            $output .= "</th>";
+        }
+
+        if ($table_row && $label_row) {
+            $output .= "</tr><tr>";
+        }
+
+        if ($table_row) {
+            $output .= "<td colspan='".($label_row ? 2 : 1)."'>";
+        }
+        $output .= $html;
+        if ($table_row) {
+            $output .= "</td>";
+        }
+        if ($table_row) {
+            $output .= "</tr>";
+        }
+
+        return $output;
+    }
```

Comment:
main.php still uses a lot of small non-overridable functions to generate tables (I think we should avoid that) but at least this one can be overridden now.

Reply:
FWIW the current approach of adding lots of TD/TH/TR elements seems flexible but messy - ideally this'd be another place where the extension just returns a list of `{setting_name, human_name, value, display_style}` and the theme takes care of HTML, but I'm not sure how to do that without massively reducing flexibility...

(This isn't something that needs fixing right away, just something that I have in the back of my mind)

Reply:
It seems flexible right until you don't want to use tables :smile: 
Also, I'm noticing extensions create their tables in very different ways, which explains why I had so much trouble getting everything to look consistent on my theme. There's even setupblocks that don't use tables at all and just populate the items in the div.

I know it's going to look a lot less flexible when I commit this. But giving extensions that much flexibility leads to inconsistencies...
I'm also trying to send everything through this `format_option` function, which I now renamed to `format_item` and takes care of labels and text as well, and I created other functions in theme.php to wrap them in the table elements (the ones one would override to get rid of tables entirely)

We can discard it if you think it's too much anyway. I'm adapting all extensions to the revised format now, I'll probably commit after that.